### DISCUSSION TITLE
confusione....

### DISCUSSION THREAD

**Discussion by bf69** [indomus/forum#1111] [HUB: Home Assistant]
buongiorno, attualmente ho HA Core su raspberry in Venv con accesso da remoto tramite Duckdns.
vorrei portare HA su Docker e siccome ho fatto un primo tentativo ma mi sono arenato su caddy, ho visto la guida per installare una VPN tramite OpenVpn. la mia domanda è: posso eliminare duckdns, caddy e utilizzare solo openvpn? conviene farlo prima di passare a docker?
grazie

**Comment by Scialla** [comment]
Tutto si può fare, dipende da cosa vuoi fare....

Con OPENVPN ti colleghi da remoto come se fossi dentro casa, quindi accederai alla tua istanza di HA con l'IP interno (192.168.x.x:8123) e non con il tuo FQDN (miacasa.duckdns.org:8123) e per farlo, ogni volta, dovrai abilitare openVPN nella macchina dalla quale ti vuoi collegare a casa..... direi che è una soluzione più indirizzata a fare eventuali modifiche o cose diverse che per collegarsi sempre alla propria istanza di HA....

**Reply by bf69** [reply]
ok, ora è più chiaro, grazie

**Reply by snakuzzo** [reply]
considera che se hai intenzione di utilizzare assistenti vocali per pilotare entità di HA, per farlo devi per forza esporre su internet HA in https

**Reply by bf69** [reply]
sto provando a seguire la guida per utilizzare caddy su docker ma sono fermo...
una volta creato il caddy file, e il contanier, il log da questo:
`pi@raspberrypi:~/home-assistant $ docker logs caddy
{"level":"info","ts":1638011163.417373,"msg":"using provided configuration","config_file":"/etc/caddy/Caddyfile","config_adapter":"caddyfile"}
{"level":"warn","ts":1638011163.420712,"msg":"input is not formatted with 'caddy fmt'","adapter":"caddyfile","file":"/etc/caddy/Caddyfile","line":3}
{"level":"info","ts":1638011163.423423,"logger":"admin","msg":"admin endpoint started","address":"tcp/localhost:2019","enforce_origin":false,"origins":["127.0.0.1:2019","localhost:2019","[::1]:2019"]}
{"level":"info","ts":1638011163.4239101,"logger":"http","msg":"server is listening only on the HTTPS port but has no TLS connection policies; adding one to enable TLS","server_name":"srv0","https_port":443}
{"level":"info","ts":1638011163.4239638,"logger":"http","msg":"enabling automatic HTTP->HTTPS redirects","server_name":"srv0"}
{"level":"info","ts":1638011163.423894,"logger":"tls.cache.maintenance","msg":"started background certificate maintenance","cache":"0x38201e0"}
{"level":"info","ts":1638011163.425088,"logger":"tls","msg":"cleaning storage unit","description":"FileStorage:/data/caddy"}
{"level":"info","ts":1638011163.4254806,"logger":"http","msg":"enabling automatic TLS certificate management","domains":["casafabio.duckdns.org"]}
{"level":"info","ts":1638011163.426618,"logger":"tls","msg":"finished cleaning storage units"}
{"level":"info","ts":1638011163.428613,"msg":"autosaved config (load with --resume flag)","file":"/config/caddy/autosave.json"}
{"level":"info","ts":1638011163.428666,"msg":"serving initial configuration"}
{"level":"info","ts":1638011726.6085546,"msg":"shutting down apps, then terminating","signal":"SIGTERM"}
{"level":"warn","ts":1638011726.6087258,"msg":"exiting; byeee!! 👋","signal":"SIGTERM"}
{"level":"info","ts":1638011726.6115847,"logger":"tls.cache.maintenance","msg":"stopped background certificate maintenance","cache":"0x38201e0"}
{"level":"info","ts":1638011726.612982,"logger":"admin","msg":"stopped previous server","address":"tcp/localhost:2019"}
{"level":"info","ts":1638011726.6130598,"msg":"shutdown complete","signal":"SIGTERM","exit_code":0}
{"level":"info","ts":1638011728.2864647,"msg":"using provided configuration","config_file":"/etc/caddy/Caddyfile","config_adapter":"caddyfile"}
{"level":"warn","ts":1638011728.2898152,"msg":"input is not formatted with 'caddy fmt'","adapter":"caddyfile","file":"/etc/caddy/Caddyfile","line":3}
{"level":"info","ts":1638011728.297279,"logger":"admin","msg":"admin endpoint started","address":"tcp/localhost:2019","enforce_origin":false,"origins":["localhost:2019","[::1]:2019","127.0.0.1:2019"]}
{"level":"info","ts":1638011728.2978692,"logger":"http","msg":"server is listening only on the HTTPS port but has no TLS connection policies; adding one to enable TLS","server_name":"srv0","https_port":443}
{"level":"info","ts":1638011728.29793,"logger":"http","msg":"enabling automatic HTTP->HTTPS redirects","server_name":"srv0"}
{"level":"info","ts":1638011728.2989564,"logger":"tls.cache.maintenance","msg":"started background certificate maintenance","cache":"0x2c98000"}
{"level":"info","ts":1638011728.2995293,"logger":"http","msg":"enabling automatic TLS certificate management","domains":["casafabio.duckdns.org"]}
{"level":"info","ts":1638011728.3026645,"msg":"autosaved config (load with --resume flag)","file":"/config/caddy/autosave.json"}
{"level":"info","ts":1638011728.3027132,"msg":"serving initial configuration"}
{"level":"info","ts":1638011728.3028111,"logger":"tls","msg":"cleaning storage unit","description":"FileStorage:/data/caddy"}
{"level":"info","ts":1638011728.3040502,"logger":"tls","msg":"finished cleaning storage units"}
`
che non è proprio quello che mi aspettavo...

**Reply by bf69** [reply]
perchè instanziando caddy alla voce volume viene definita la posizione del caddyfile come "/etc/caddy" e non "/home/pi/caddy/"  che è la posizione dove viene creato?
` volumes:
    - "/home/pi/caddy/Caddyfile:/etc/caddy/Caddyfile"
    - "/home/pi/caddy/data:/data"`
mi sfugge qualcosa....

**Reply by snakuzzo** [reply]
Sì...ti sfugge come funziona Docker...e prima di continuare ti consiglio di andare a guardarti come funziona il mount dei volumi.
La guida che abbiamo scritto non è un manuale di istruzioni per chi non conosce Docker.
Io posso pure spiegarti qui il perché, ma...

1. se non sbaglio è stato accennato nelle nostre guide
2. non siamo noi a dover spiegare il funzionamento di Docker
3. faresti molto prima andando a leggere la documentazione ufficiale [QUI](https://docs.docker.com/storage/volumes/) che di sicuro è più precisa

**Reply by bf69** [reply]
sono due anni che utilizzo HA e mi sono fatto un pochino di esperienza leggendo le vostre guide che sono molto precise e qualcos'altro in rete. Adesso per vari motivi conviene usare Docker e sono un pò spiazzato, ho bisogno di tempo, pensavo che il passaggio fosse meno difficile, anche se tutto sommato mi sono rimasti il problema della chiavetta USB e Caddy. Il primo andrò a documentarmi, per Caddy ho seguito la guida alla lettera ma non ne esco....sicuramente sbaglierò qualcosa...
per adesso grazie per la pazienza

**Reply by snakuzzo** [reply]
Il problema è proprio questo...seguire alla lettera. Se lo fai devi almeno sapere dei concetti base

**Reply by bf69** [reply]
a furia di picchiare la testa alla fine ci sono riuscito!
non avevo  inserito Docker al gruppo sudoers:
`sudo groupadd docker
sudo usermod -aG docker $USER
sudo reboot`
adesso rimane la chiavetta Usb da montare come volume e alexa che non rileva più i dispositivi esposti.
con calma ci arrivo....
